components/raftstore-v2/src/operation/command/mod.rs

```diff
@@ -451,18 +481,7 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
 
     #[inline]
     pub fn flush(&mut self) {
-        if let Some(wb) = self.write_batch_mut() && !wb.is_empty() {
-            let mut write_opt = WriteOptions::default();
-            write_opt.set_disable_wal(true);
-            if let Err(e) = wb.write_opt(&write_opt) {
-                panic!("failed to write data: {:?}", self.logger.list());
-            }
-            if wb.data_size() <= APPLY_WB_SHRINK_SIZE {
-                wb.clear();
-            } else {
-                self.write_batch_mut().take();
-            }
-        }
+        self.flush_write();
```

Comment:
Why the change?

Reply:
Before creating checkpoint, the write batch should be flushed. Extract this so that exec_batch_split can use it.